File: app/src/main/java/bffisher/fasnap/MainActivity.java
package bffisher.fasnap;

import android.app.Activity;
import android.app.Fragment;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.BottomNavigationView;
import android.view.MenuItem;

public class MainActivity extends Activity implements OnFragmentInteractionListener{
    private Fragment home;
    private Fragment edit;

    private BottomNavigationView.OnNavigationItemSelectedListener mOnNavigationItemSelectedListener
            = new BottomNavigationView.OnNavigationItemSelectedListener() {

        @Override
        public boolean onNavigationItemSelected(@NonNull MenuItem item) {
            switch (item.getItemId()) {
                case R.id.navigation_home:
                    getFragmentManager().beginTransaction()
                            .replace(R.id.fragment_container, home).commit();
                    break;
                case R.id.navigation_edit:
                    getFragmentManager().beginTransaction()
                            .replace(R.id.fragment_container, edit).commit();
                    break;
            }
            item.setChecked(true);
            return false;
        }
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        BottomNavigationView navigation = (BottomNavigationView) findViewById(R.id.navigation);
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener);

        home = HomeFragment.newInstance();
        edit = EditFragment.newInstance(10);

        getFragmentManager().beginTransaction()
                .replace(R.id.fragment_container, home).commit();
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }
}

File: app/src/main/java/bffisher/fasnap/MainActivity.kt
package bffisher.fasnap

import android.app.Activity
import android.app.Fragment
import android.net.Uri
import android.os.Bundle
import android.support.design.widget.BottomNavigationView
import bffisher.fasnap.edit.MainFragment
import bffisher.fasnap.edit.OnFragmentInteractionListener

class MainActivity : Activity(), OnFragmentInteractionListener {
    private var home: Fragment? = null
    private var edit: Fragment? = null

    private val mOnNavigationItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { item ->
        when (item.itemId) {
            R.id.navigation_home -> fragmentManager.beginTransaction()
                    .replace(R.id.main_container, home).commit()
            R.id.navigation_edit -> fragmentManager.beginTransaction()
                    .replace(R.id.main_container, edit).commit()
        }
        item.isChecked = true
        false
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout._main)

        val navigation = findViewById<BottomNavigationView>(R.id.navigation)
        navigation.setOnNavigationItemSelectedListener(mOnNavigationItemSelectedListener)

        home = HomeFragment.newInstance()
        edit = MainFragment.newInstance()

        fragmentManager.beginTransaction().replace(R.id.main_container, home).commit()
    }

    override fun onFragmentInteraction(uri: Uri) {

    }
}
